trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid/module_persistence_levelpool_hybrid_state.F

```diff
@@ -0,0 +1,76 @@
+! This module defines and instantiates objects
+! for a hybrid type reservoir's state.
+! State holds and tracks dynamic/changing variables
+! that are only relevant to the given hybrid
+! reservoir object and not other modules or areas
+! of the system.
+
+#define seconds_in_day 86400
+
+module module_persistence_levelpool_hybrid_state
+
+    use module_levelpool
+    use module_reservoir_base
+    implicit none
+
+    ! Extend/derive hybrid state from the abstract base
+    ! struct for reservoir state.
+    type, extends(reservoir_base_state_struct) :: hybrid_state_interface
```

Comment:
is interface the extension of a struct?
can we just call it hybrid_state?

Reply:
maybe an interface is the extension of an empty abstract class?
I have to assume that at some point the abstract class may be changed to have some variables.
It's just really odd to see every necessary variable added in this extension.

Reply:
why not `type, extends(levelpool_state_interface) :: hybrid_state` ?
That would seem to make it more of a hybrid. Is it because you need separate init and destroy?

Reply:
type would be fine for most of these. struct was orginaly used to indicate data structures that would not have bound functions such as the Overland and Subsurface data structures, interface was denoted a pure virtual base class, anything else should probably be \<module_name\>_type.

Reply:
@jdmattern-noaa fixing. Changed 'struct' to 'type'. Keeping 'interface' per Donald's explanation.